sklearn/datasets/base.py

```diff
@@ -468,14 +478,17 @@ def load_digits(n_class=10):
         flat_data, target = flat_data[idx], target[idx]
         images = images[idx]
 
+    if return_X_y:
+        return flat_data, target.astype(np.int)
```

Comment:
I think the `astype` should precede this branching. Even do it right after the file reading block.